Adds a management command to create discount code(s) via command line

jkachel opened the pull request:
#### Pre-Flight checklist

- [X] Testing
  - [X] Code is tested
  - [X] Changes have been manually tested

#### What are the relevant tickets?

#1139 

#### What's this PR do?

Adds a command to create discount/enrollment codes via the command line. Optionally, it can create batches of codes.

#### How should this be manually tested?

Run the management command `generate_discount_code --help` to see the options, then test with various permutations:

* The defaults are to create a `percent-off` code with unlimited redemptions for the specified amount with the code specified. (You can specify multiple codes on the command line to generate several codes.) 
* You can add an expiration date by specifying `--expires`, or make the code a one-time redemption code by using `--one-time`
* The discount type can be specified with `--discount-type`
* Use `--prefix` and `--count` to generate a batch of codes. In this mode, the code specified on the command line will be ignored, and it will instead use your specified prefix and a UUID as the code. 

The command will write its output to `generated-codes.csv`. This file will be overwritten on each run.

asadali145 commented on ecommerce/management/commands/generate_discount_code.py:
```diff
@@ -0,0 +1,175 @@
+        )
+
+        parser.add_argument(
+            "codes",
```
IMO we should add a bit more detail about using codes. Maybe in the module docs or here about using codes.

asadali145 reviewed (changes requested):
Works fine, Mentioned a few things that I noticed in this.

asadali145 commented:
Also FYI @jkachel, I am not sure about that 2 years thing mentioned in slack and in the ticket `that expire in two years` as well.

jkachel commented:
Addressed comments in https://github.com/mitodl/mitxonline/pull/1142/commits/3272508579256d3ba7d9c2dc3facd261ae3ea77c . The expiration date can be set with `--expires`, so to complete the ticket I would just specify the expiration date on the command line.

asadali145 reviewed (approved):
LGTM! Awesome work...
